subsys/net/lib/app/net_app.c

```diff
@@ -567,15 +569,18 @@ static struct net_context *get_server_ctx(struct net_app_ctx *ctx,
 
 			if (net_ipv6_addr_cmp(addr6, remote6) &&
 			    port == rport) {
-				NET_DBG("Selecting net_ctx %p for AF_INET6 "
-					"port %d", tmp, ntohs(rport));
+				NET_DBG("Selecting net_ctx %p iface %p for "
+					"AF_INET6 port %d", tmp,
+					net_context_get_iface(tmp),
+					ntohs(rport));
 				return tmp;
 			}
 		}
 
 		if (IS_ENABLED(CONFIG_NET_IPV4) &&
-		    tmp->remote.sa_family == AF_INET &&
-		    dst->sa_family == AF_INET) {
+		    (dst->sa_family == AF_UNSPEC ||
```

Comment:
Ah.. I mentioned this in the other patch.   This change should be moved there.

Reply:
+1

Reply:
Yes, I made a mistake when rebasing the set. This definitely belongs to the other patch.